### ISSUE TITLE
Issue: Discord Rich Presence doesn't work again (version 1.6.1) and since the 1.5.3

### ISSUE THREAD

**Issue by Nebelous555** [pizzaboxer/bloxstrap#84]
![image](https://user-images.githubusercontent.com/120223163/218861448-095c6d09-ff21-4da5-a7a5-d4df5d079775.png)
![image](https://user-images.githubusercontent.com/120223163/218861494-a71e00cf-b7f9-42f3-868e-a70a8431d9de.png)
![image](https://user-images.githubusercontent.com/120223163/218861708-cfcbd65b-10b8-4dff-867a-8ee9fd57acb0.png)
![image](https://user-images.githubusercontent.com/120223163/218861747-eafdd7cc-9a8e-47d4-aec3-9054e201a8f0.png)

**Comment by pizzaboxer** [comment]
You mentioned that it "doesn't work again". Was it working fine before?

**Comment by Nebelous555** [comment]
Nope, it was meant to work with new update, but again with update, it's not working whatever you update.

**Comment by pizzaboxer** [comment]
I want to quickly narrow down this issue to see if it's a problem with Bloxstrap or something else. I just quickly made a test program, run it and see if a test presence appears.

[https://pizzaboxer.xyz/files/DiscordRPCTest.exe](https://pizzaboxer.xyz/files/DiscordRPCTest.exe)

**Comment by Nebelous555** [comment]
> I want to quickly narrow down this issue to see if it's a problem with Bloxstrap or something else. I just quickly made a test program, run it and see if a test presence appears.
> 
> https://pizzaboxer.xyz/files/DiscordRPCTest.exe

![image](https://user-images.githubusercontent.com/120223163/219410399-973a972c-9e10-4ca3-a640-cab054b423fb.png)

**Comment by pizzaboxer** [comment]
That seems to clear things up a little bit. With Discord open, go to [http://localhost:6463](http://localhost:6463) in your browser. What shows when you go there?

**Comment by Nebelous555** [comment]
> That seems to clear things up a little bit. With Discord open, go to http://localhost:6463 in your browser. What shows when you go there?

![image](https://user-images.githubusercontent.com/120223163/219416811-f3e5d3c7-620a-4142-ab17-188ce35d7670.png)

**Comment by pizzaboxer** [comment]
Strange. Do any other programs that use Discord Rich Presence work?

**Comment by Nebelous555** [comment]
> Strange. Do any other programs that use Discord Rich Presence work?

I have one that work, but it requires my security cookie to fetch data from API to set presence. Others use cookie code too
![image](https://user-images.githubusercontent.com/120223163/219421738-f75c4194-8aa5-43eb-ba15-4bfc18ad5ba7.png)

**Comment by pizzaboxer** [comment]
Which one is this specifically? Just checking to make sure.

**Comment by Nebelous555** [comment]
> Which one is this specifically? Just checking to make sure.

https://devforum.roblox.com/t/stateye-discord-rich-presence-for-roblox/2052599

**Comment by pizzaboxer** [comment]
I'm even more confused because the way I do rich presence shouldn't work any differently to this. Made the logs more detailed with the test app, run it and tell me what it says. It's going to be a lot more, so just copy paste whatever it shows.

[https://pizzaboxer.xyz/files/DiscordRPCTest2.exe](https://pizzaboxer.xyz/files/DiscordRPCTest2.exe)

**Comment by Nebelous555** [comment]
> I'm even more confused because the way I do rich presence shouldn't work any differently to this. Made the logs more detailed with the test app, run it and tell me what it says. It's going to be a lot more, so just copy paste whatever it shows.
> 
> https://pizzaboxer.xyz/files/DiscordRPCTest2.exe


it says:

"
Initializing...
INFO: Attempting a new connection
Initialized
Setting presence...
INFO: RPC Connection Started
TRACE: ============================
TRACE: Assembly:             DiscordRPC, Version=1.1.3.18, Culture=neutral, PublicKeyToken=null
TRACE: Enqueue Command: DiscordRPC.RPC.Commands.PresenceCommand
TRACE: Pipe:                 DiscordRPC.IO.ManagedNamedPipeClient
Presence set
TRACE: Platform:             Microsoft Windows NT 10.0.19044.0
TRACE: applicationID:        1005469189907173486
TRACE: targetPipe:           -1
TRACE: POLL_RATE:            1000
TRACE: _maxRtQueueSize:      512
TRACE: _maxRxQueueSize:      0
TRACE: ============================
TRACE: Connecting to the pipe through the DiscordRPC.IO.ManagedNamedPipeClient
TRACE: ManagedNamedPipeClient.Connection(-1)
TRACE: Connection Attempt 0 ()
INFO: Attempting to connect to 'discord-ipc-0'
ERR : Failed connection to discord-ipc-0. Access to the path is denied.
WARN: Tried to close a already closed pipe.
TRACE: Done. Result: True
TRACE: Skipping sandbox connection.
TRACE: Connection Attempt 1 ()
INFO: Attempting to connect to 'discord-ipc-1'
ERR : Failed connection to discord-ipc-1. The operation has timed out.
WARN: Tried to close a already closed pipe.
TRACE: Done. Result: True
TRACE: Skipping sandbox connection.
TRACE: Connection Attempt 2 ()
INFO: Attempting to connect to 'discord-ipc-2'
ERR : Failed connection to discord-ipc-2. The operation has timed out.
WARN: Tried to close a already closed pipe.
TRACE: Done. Result: True
TRACE: Skipping sandbox connection.
"

**Comment by Nebelous555** [comment]
> What about if you run the test app as Administrator?

"
Initializing...
INFO: Attempting a new connection
Initialized
Setting presence...
INFO: RPC Connection Started
TRACE: ============================
TRACE: Assembly:             DiscordRPC, Version=1.1.3.18, Culture=neutral, PublicKeyToken=null
TRACE: Pipe:                 DiscordRPC.IO.ManagedNamedPipeClient
TRACE: Platform:             Microsoft Windows NT 10.0.19044.0
TRACE: applicationID:        1005469189907173486
TRACE: Enqueue Command: DiscordRPC.RPC.Commands.PresenceCommand
Presence set
TRACE: targetPipe:           -1
TRACE: POLL_RATE:            1000
TRACE: _maxRtQueueSize:      512
TRACE: _maxRxQueueSize:      0
TRACE: ============================
TRACE: Connecting to the pipe through the DiscordRPC.IO.ManagedNamedPipeClient
TRACE: ManagedNamedPipeClient.Connection(-1)
TRACE: Connection Attempt 0 ()
INFO: Attempting to connect to 'discord-ipc-0'
TRACE: Waiting for connection...
INFO: Connected to 'discord-ipc-0'
TRACE: Done. Result: False
TRACE: Begining Read of 16384 bytes
TRACE: Connected to the pipe. Attempting to establish handshake...
Established connection with Discord RPC
TRACE: Enqueued Message, but queue size is 0.
TRACE: Attempting to establish a handshake...
TRACE: Sending Handshake...
TRACE: Setting the connection state to CONNECTING
TRACE: Connection Established. Starting reading loop...
TRACE: Ending Read
TRACE: Read 358 bytes
TRACE: Read a frame: Frame
TRACE: Starting another read
TRACE: Begining Read of 16384 bytes
TRACE: Read Payload: Frame
INFO: Handling Response. Cmd: Dispatch, Event: Ready
INFO: Connection established with the RPC
TRACE: Setting the connection state to CONNECTED
Received ready from user Neb (my id or smth ) (was 4851+ etc)
TRACE: Enqueued Message, but queue size is 0.
TRACE: Attempting to send payload: SetActivity
TRACE: Sending payload: SetActivity
TRACE: Sent Successfully.
TRACE: Ending Read
TRACE: Read 197 bytes
TRACE: Read a frame: Frame
TRACE: Starting another read
TRACE: Begining Read of 16384 bytes
TRACE: Read Payload: Frame
INFO: Handling Response. Cmd: SetActivity, Event:
Updated presence
TRACE: Enqueued Message, but queue size is 0.
"

**Comment by pizzaboxer** [comment]
Is it showing on your Discord activity?

**Comment by Nebelous555** [comment]
maybe i need to run bloxstrap as admin idk

**Comment by Nebelous555** [comment]
> nvm it did, solved

![image](https://user-images.githubusercontent.com/120223163/219477206-96ad7b2d-64eb-4cdf-8f8c-763caabb0bd6.png)

**Comment by pizzaboxer** [comment]
Before I finish here, I have a feeling you're running Discord as Administrator.
Open Task Manager, go to the Details tab, right-click on the 'Name' header, select 'Select Columns', find 'Elevated' and make sure it's checked.
![image](https://user-images.githubusercontent.com/41478239/219477714-5da56c06-3274-4180-8794-14698f07d3b3.png)
Find Discord in the Details list. What does it say under the Elevated column?

**Comment by Nebelous555** [comment]
> Before I finish here, I have a feeling you're running Discord as Administrator. Open Task Manager, go to the Details tab, right-click on the 'Name' header, select 'Select Columns', find 'Elevated' and make sure it's checked. ![image](https://user-images.githubusercontent.com/41478239/219477714-5da56c06-3274-4180-8794-14698f07d3b3.png) Find Discord in the Details list. What does it say under the Elevated column?

Yeah it runs as Elevated, now when I run Bloxstrap as admin, it works on discord with and without admin, strange.

**Comment by pizzaboxer** [comment]
It was failing to set your presence because Discord was running as admin. Not sure why it was, but hopefully you can figure that out.
